solc/CommandLineInterface.cpp

```diff
@@ -730,12 +730,15 @@ map<string, Json::Value> CommandLineInterface::parseAstFromInput()
 void CommandLineInterface::createFile(string const& _fileName, string const& _data)
 {
 	namespace fs = boost::filesystem;
-	// create directory if not existent
-	fs::path p(m_args.at(g_argOutputDir).as<string>());
-	// Do not try creating the directory if the first item is . or ..
-	if (p.filename() != "." && p.filename() != "..")
```

Comment:
The problem here was that `filename()` is just the last segment of the path so it returns `.` not only for literal `.` but also for paths with empty file name, e.g. ones ending with a slash.